packages/common/src/tools/LogThrottler.ts

```diff
@@ -0,0 +1,86 @@
+// eslint-disable-next-line import/no-cycle
+import { Logger, LoggerOptions } from './Logger'
+
+interface LogInfo {
+  count: number
+  throttleCount: number
+  isThrottling: boolean
+}
+
+export interface LogThrottlerOptions {
+  readonly threshold: number
+  readonly thresholdTimeInMs: number
+  readonly throttleTimeInMs: number
+}
+/**
+ * @summary Aim of this class is to reduce log output by throttling.
+ *
+ * @description It allows to throttle given log for throttleTimeInMs if it was called threshold times in throttleTimeInMs
+ * by storing count of how much time it was called in recentLogs by logKey.
+ */
+export class LogThrottler {
+  private readonly recentLogs: Map<string, LogInfo>
+  private readonly logger: Logger
+
+  constructor(
+    private readonly options: LogThrottlerOptions,
+    loggerOptions: LoggerOptions,
+  ) {
+    this.recentLogs = new Map<string, LogInfo>()
+    this.logger = new Logger(loggerOptions).for(LogThrottler.name)
```

Comment:
Why can't we just inject it as parameter? in application.ts we could create a "fresh" logger just for this class but in tests we could easily mock it.

Reply:
We can do a separate logger in application.ts just for LogThrottler, I was thinking about it but it would look kinda ugly.

Reply:
I think it's fine -- this is really only sane way to test it. Dependency Injection all the way down :D